stan/math/prim/fun/as_value_column_array_or_scalar.hpp

```diff
@@ -0,0 +1,105 @@
+#ifndef STAN_MATH_PRIM_FUN_AS_VALUE_COLUMN_ARRAY_OR_SCALAR
+#define STAN_MATH_PRIM_FUN_AS_VALUE_COLUMN_ARRAY_OR_SCALAR
+
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/meta.hpp>
+#include <vector>
+
+namespace stan {
+namespace math {
+
+/**
+ * Extract values from var types and containers and return a column vector or a
+ * scalar. For arithmetic scalar inputs that is an identity function.
+ *
+ * @tparam T Type of scalar element.
+ * @param a Specified scalar.
+ * @return the scalar.
+ */
+template <typename T, require_stan_scalar_t<T>* = nullptr>
+inline auto as_value_column_array_or_scalar(const T& a) {
+  return value_of(a);
+}
+
+/**
+ * Extract values from var types and containers and return a column vector or a
+ * scalar. For arithmetic column vector inputs this is an identity function.
+ *
+ * @tparam T Type of scalar element.
+ * @param a Specified vector.
+ * @return Same vector.
+ */
+template <typename T, require_col_vector_t<T>* = nullptr,
+          require_not_st_arithmetic<T>* = nullptr>
+inline auto as_value_column_array_or_scalar(T&& a) {
+  return value_of(std::forward<T>(a)).array();
+}
+
+/**
+ * Extract values from var types and containers and return a column vector or a
+ * scalar. For arithmetic column vector inputs this is an identity function.
+ *
+ * @tparam T Type of scalar element.
+ * @param a Specified vector.
+ * @return Same vector.
+ */
+template <typename T, require_col_vector_t<T>* = nullptr,
+          require_st_arithmetic<T>* = nullptr>
+inline auto as_value_column_array_or_scalar(T&& a) {
+  return std::forward<T>(a).array();
+}
+
+/**
+ * Extract values from var types and containers and return a column vector or a
+ * scalar. For arithmetic row vector input this is transpose.
+ *
+ * @tparam T Type of scalar element.
+ * @param a Specified vector.
+ * @return Transposed vector.
+ */
+template <typename T, require_row_vector_t<T>* = nullptr,
+          require_not_col_vector_t<T>* = nullptr,
+          require_not_st_arithmetic<T>* = nullptr>
+inline auto as_value_column_array_or_scalar(T&& a) {
+  return value_of(a.transpose()).array();
```

Comment:
Since `value_of()` for non-arithmetic column vectors goes to the `value_of()` that makes a holder do we need one here? Like if a function only calls another function that uses a holder does it still need a holder?

https://github.com/stan-dev/math/blob/develop/stan/math/prim/fun/value_of.hpp#L53

Reply:
Yes, unfortunately we do. `value_of` will only store the transpose expression in the holder. As you have written the code we would in fact need two more holders - on for `a` and one for the `value_of` return value. But you can pull transpose out of `value_of`. Than you only need one for `value_of` return value.

Reply:
Actually I was wrong about two holders. `value_of` return value is an expression so it does not need another holder. But you still need one for `a`.